src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs

```diff
@@ -64,7 +64,8 @@ public override void ExecuteCmdlet()
                     if (this.VirtualMachineScaleSet != null
                             && this.VirtualMachineScaleSet.VirtualMachineProfile != null
                             && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile != null
-                            && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference != null)
+                            && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference != null
+                            && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference.Id != null)
```

Comment:
Could you describe the issue that caused this bug and how this edit fixes it?

Reply:
The issue is here @adam https://github.com/Azure/azure-powershell/issues/22195

The IF condition nulls the imagereference even when imagereference is being updated with normal rbac based ARM image using Imagereference.ID

It's pretty straightforward, imagereference.Id is passed for a new image, but the old IF condition did not check for that property and went into the Condition nullifiying imagereferece. Hence User's SKU was not updated.

I've already tested the fix. Let me know if there is further explanation needed.